File: backend/src/main/java/com/airassist/backend/model/Comment.java
package com.airassist.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;
import java.util.UUID;

@Entity
@Table(name = "comments")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(nullable = false)
    private User user;

    @Column(nullable = false, length = 1000)
    private String text;

    @Column(nullable = false)
    private Timestamp timestamp;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(nullable = false, referencedColumnName = "id")
    private Case caseEntity;
}

File: frontend/src/app/shared/types/comment.ts
import { User } from './user';
import { Timestamp } from 'rxjs';
import { Case } from './case';

export type Comment = {
  id?: string;
  user: User;
  text: string;
  timestamp: string; // ISO string format: "2025-08-04T10:30:00.000Z"
  caseEntity: Case;
};
